### PR TITLE
[SPIKE] [CFP-403] FormBuilder migration- Fixed fees

### PR DESCRIPTION
**Pull request by imtiazAhmd**
#### What
Migrate external_users form step: fixed_fees form

#### Ticket
[FormBuilder migration - Fixed fees](https://dsdmoj.atlassian.net/browse/CFP-403)

#### Why

- Update the service to use GOV.UK styles, components and patterns
- To improve and resolve the service's accessibility issues
- To reduce code complexity
- To improve maintainability

#### TODO (wip)

 - [X] Advocate partial
 - [x] Litigator partial
 - [x] Fix Cucumber Failing tests
 - [x] Fix rSpecs Failing tests

### CODE REVIEW

**Comment by imtiazAhmd** [comment]
Closing pull request as SPIKE now has been completed. We can revisit this branch when working on the Implementation tickets.

File: features/step_definitions/common_steps.rb
Comment (naseberry) [inline features/step_definitions/common_steps.rb]: As this is a common step definition, this may fail on features that test unmigrated form steps

File: app/views/external_users/advocates/_advocate_category.html.haml
Comment (naseberry) [inline app/views/external_users/advocates/_advocate_category.html.haml:1]: Missing css class name 'advocate-categories' for input

File: app/views/external_users/claims/_date_attended_fields.html.haml
Comment (naseberry) [inline app/views/external_users/claims/_date_attended_fields.html.haml]: Nit: is `id` redundant?

File: app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml
Comment (naseberry) [inline app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml]: Nit: are we migrating these `hidden_field` helpers to the relevant design system helper?

File: app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml
Comment (naseberry) [inline app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml]: Nit: discuss using the summary list modifier `govuk_summary_list_no_border`

File: app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml
Comment (naseberry) [inline app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml:66]: Nit: are we migrating these `hidden_field` helpers to the relevant design system helper?

File: app/validators/fee/base_fee_validator.rb
Comment (naseberry) [inline app/validators/fee/base_fee_validator.rb]: Nit: invalid string

File: app/views/external_users/claims/fixed_fees/litigators/_fields.html.haml
Comment (naseberry) [inline app/views/external_users/claims/fixed_fees/litigators/_fields.html.haml]: Is the `tag` option required?

File: config/locales/en.yml
Comment (naseberry) [inline config/locales/en.yml]: Nit: could we change this value to use the `date_hint` variable

File: config/locales/en/error_messages/claim.yml
Comment (naseberry) [inline config/locales/en/error_messages/claim.yml]: Nit: could we replace all the relevant `short` values

File: config/locales/en/models/fixed_fee.yml
Comment (naseberry) [inline config/locales/en/models/fixed_fee.yml]: Can we discuss moving this to `config/locales/en/models/claim.yml`

File: features/fee_calculator/advocate/graduated_fee_calculator.feature
Comment (naseberry) [inline features/fee_calculator/advocate/graduated_fee_calculator.feature]: Nit: remove comment

File: app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml
Comment (imtiazAhmd) [reply app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml]: its been already fixed in later commit
https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/4730/commits/63063b29bbf9322b2ddefed2f3f2dc37ac4a8cc5

File: app/validators/fee/base_fee_validator.rb
Comment (imtiazAhmd) [reply app/validators/fee/base_fee_validator.rb]: its been fixed in later commit: https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/commit/63063b29bbf9322b2ddefed2f3f2dc37ac4a8cc5

File: config/locales/en/error_messages/claim.yml
Comment (imtiazAhmd) [reply config/locales/en/error_messages/claim.yml]: Already fixed in later commit https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/commit/0db7ed5c8d72af00962f41da70a5a0bde6a40b7c

Also the key has been change to `fixed_fee_date_cannot_be_too_far_in_the_future:`

File: config/locales/en.yml
Comment (imtiazAhmd) [reply config/locales/en.yml]: Already fixed in later commit https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/commit/a6d0e4c257d8d97ae14ce22ea7e09554dc1e4ab7

File: config/locales/en/models/fixed_fee.yml
Comment (imtiazAhmd) [reply config/locales/en/models/fixed_fee.yml]: Done already in a later commit
https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/commit/a6d0e4c257d8d97ae14ce22ea7e09554dc1e4ab7

File: app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml
Comment (imtiazAhmd) [reply app/views/external_users/claims/fixed_fees/advocates/_fixed_fee_fields.html.haml:66]: Changing these to the design system helper fails some cucumber tests due to how the ID has been assigned. keeping this as is, may be we can have a session together to sort this out.

File: spec/validators/date_attended_validator_spec.rb
Comment (naseberry) [inline spec/validators/date_attended_validator_spec.rb:22]: **Nit**

_Coding standard_: Line length seems really long, linting has not erred because file is being ignored by Rubocop.

If time permits, it would be nice to remove this file from Rubocop's ignore list and correct the line length issue

File: spec/validators/fee/base_fee_validator_spec.rb
Comment (naseberry) [inline spec/validators/fee/base_fee_validator_spec.rb]: _Suggestion_: Matcher improvements

`.* fees must not have a rate`
OR
`(.*?)(Pages of prosecution evidence|Number of prosecution witnesses) fees must not have a rate`

**Comment by naseberry** [comment]
AGFS missing translations, see attached

Fixed_Fee_AGFS

**Comment by naseberry** [comment]
AGFS translations issues have been posted to the ticket and no issues have been identified for LGFS

File: spec/validators/date_attended_validator_spec.rb
Comment (imtiazAhmd) [reply spec/validators/date_attended_validator_spec.rb:22]: I would like to leave this for now and will look into it, if time permits. If not, may be we can add this task to our ENDGAME?

**Comment by naseberry** [comment]
**Migration note**:

Some partials, form attribute names, validators are shared across multiple form steps (i.e. `:advocate_category` etc.)

Un-migrated form steps (i.e. basic, disbursements, misc etc.) will temporarily experience unlinked error summary links until form steps are migrated.

This is due to migrated form helpers/inputs being used with the old error summary instead of the new govuk_error_summary, see: `app/views/external_users/claims/_error_summary.html.haml`

**Review by naseberry — APPROVED** [review]
👍  LGTM, although, I would recommend a second review & approval before merging

File: app/presenters/fee/base_fee_presenter.rb
Comment (mpw5) [inline app/presenters/fee/base_fee_presenter.rb:61]: Nitpick: could we use the translation text here instead of hardcoding? 

eg `fee.errors[:quantity].include?(I18n.t(en.<path to translation>))`

This would mean that if the error wording changes in future we wouldn't need to update this file.

**Review by mpw5 — APPROVED** [review]
Great work 🎉 

One minor nitpick, but not a blocker to releasing this.

File: app/presenters/fee/base_fee_presenter.rb
Comment (imtiazAhmd) [reply app/presenters/fee/base_fee_presenter.rb:61]: Thanks @mpw5 for your suggestion. this is not a single instance and a common pattern throughout the code base. I would like to discuss with the team and get a decision on how should we proceed to get a uniform coding practice. Thanks again and we will address this before or within the [ENDGAME](https://dsdmoj.atlassian.net/browse/CFP-358)

**Pull request merged**
